File: src/factory.ts
/**
 * 기본 훅 파이프라인 조립(팩토리).
 */
import { HookPipeline } from "./hooks/pipeline.js";
import { LoggingHook } from "./hooks/hook.js";

export function createDefaultHookPipeline(): HookPipeline {
  return new HookPipeline([new LoggingHook()]);
}

File: src/dbgit/factory.py
"""애플리케이션 기본 조립(훅 파이프라인 등)."""

from __future__ import annotations

from .hooks import HookPipeline, LoggingHook


def create_default_hook_pipeline() -> HookPipeline:
    """CLI·UI에서 공통으로 쓰는 기본 훅 파이프라인."""
    return HookPipeline([LoggingHook()])
